core/src/services/onedrive/backend.rs

```diff
@@ -145,14 +142,147 @@ impl Accessor for OnedriveBackend {
+            let decoded_response = serde_json::from_slice::<OnedriveGetItemBody>(&bytes)
+                .map_err(new_json_deserialize_error)?;
+
+            let entry_mode: EntryMode = match decoded_response.item_type {
+                ItemType::Folder { .. } => EntryMode::DIR,
+                ItemType::File { .. } => EntryMode::FILE,
+            };
+
+            let mut meta = Metadata::new(entry_mode);
+            meta.set_etag(&decoded_response.e_tag);
+
+            let last_modified = decoded_response.last_modified_date_time;
+            let date_utc_last_modified = parse_datetime_from_rfc3339(&last_modified)?;
+            meta.set_last_modified(date_utc_last_modified);
+
+            meta.set_content_length(decoded_response.size);
+
+            Ok(RpStat::new(meta))
+        } else {
+            match status {
+                StatusCode::NOT_FOUND if path.ends_with('/') => {
+                    Ok(RpStat::new(Metadata::new(EntryMode::DIR)))
+                }
+                _ => Err(parse_error(resp).await?),
+            }
+        }
+    }
+
+    /// Delete operation
+    /// Documentation: https://learn.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_delete?view=odsp-graph-online
+    async fn delete(&self, path: &str, _: OpDelete) -> Result<RpDelete> {
+        let resp = self.onedrive_delete(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::NO_CONTENT | StatusCode::NOT_FOUND => Ok(RpDelete::default()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn list(&self, path: &str, _op_list: OpList) -> Result<(RpList, Self::Pager)> {
+        let pager: OnedrivePager = OnedrivePager::new(self.root.clone(), path.into(), self.clone());
+
+        Ok((RpList::default(), pager))
+    }
+
+    async fn create_dir(&self, path: &str, _: OpCreateDir) -> Result<RpCreateDir> {
+        let path = build_rooted_abs_path(&self.root, path);
+        let path_before_last_slash = std::path::Path::new(&path)
+            .parent()
+            .ok_or_else(|| Error::new(ErrorKind::Unexpected, "invalid path"))?
+            .to_str()
+            .unwrap_or(&path);
+        let encoded_path = percent_encode_path(path_before_last_slash);
+
+        let uri = format!(
+            "https://graph.microsoft.com/v1.0/me/drive/root:{}:/children",
+            encoded_path
+        );
+
+        let folder_name = get_basename(&path);
+        let folder_name = folder_name.strip_suffix('/').unwrap_or(folder_name);
+
+        let body = serde_json::json!({
+            "name": folder_name,
+            "folder": {},
+            "@microsoft.graph.conflictBehavior": "replace"
+        });
+        let body_bytes = serde_json::to_vec(&body).map_err(new_json_serialize_error)?;
+        let async_body = AsyncBody::Bytes(bytes::Bytes::from(body_bytes));
+
+        let response = self
+            .onedrive_post(&uri, async_body, Some("application/json"))
```

Comment:
Please don't provide a general `post` API, instead use `onedrive_create_dir` and implement request build logic inside it.

Reply:
fixed